### PR TITLE
Update LSP12 Issued Asset Guides

### PR DESCRIPTION
**Pull request by fhildeb**
## PR Features:

- Adds dynamic asset encoding to the issued asset guides (separate menu for setting and updating)
- Adds description for referencing assets on a UP
- Links to new playground snippets
- Simplifies heading + adds Emoji

## Discussion

@b00ste We currently have two pages for issued assets that are almost similar (90%). One is for dApp developers, and one is the backend- the only difference being how to interact (either using an extension request or private key) 

WDYT about keeping only the dApp guide? The code within `lukso-playground` uses a private key anyway. Experts will also know how to exchange the extension provider for a private key. IMO, it's nothing LUKSO-related.

## Related Playground Update

- https://github.com/lukso-network/lukso-playground/pull/56

### CODE REVIEW

File: docs/learn/dapp-developer/register-lsp12-issued-assets.md [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
 import TabItem from '@theme/TabItem';
 
-# Set LSP12 Issued Assets
+# Register Issued Assets
 
-:::info Code Examples
+:::success Asset Authenticity
 
-The full code of this example can be found in the 👾 [lukso-playground](https://github.com/lukso-network/lukso-playground/blob/main/universal-profile/register-asset-backend.ts) repository.
+During the [creation of an asset](./deploy-contracts.md), Universal Profiles can be referenced as creators within the [LSP4 Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata.md). To **authenticate the creators** of an asset, the corresponding Universal Profiles are required to also reference the asset using [LSP12 Issued Assets](../../standards/universal-profile/lsp12-issued-assets.md). Services can then validate the **on-chain cross-link** to prove authenticity.
```
Comment (CJ42) [inline docs/learn/dapp-developer/register-lsp12-issued-assets.md:14]:
```suggestion
During the [creation of an asset](./deploy-contracts.md), Universal Profiles can be referenced as creators within the [LSP4 Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata.md). To **authenticate the creators** of an asset, their Universal Profiles must list the asset's address under the [LSP12 Issued Assets](../../standards/universal-profile/lsp12-issued-assets.md). Services can then validate the **on-chain cross-link** to prove authenticity.
```

File: docs/learn/dapp-developer/register-lsp12-issued-assets.md [hunk 2]
```diff
@@ -125,17 +125,35 @@ const issuedAssets = [
...
+const accounts = await web3.eth.getAccounts();
 ```
 
   </TabItem>
 </Tabs>
 
 ## Encode the data keys
 
-Next, you can use the `erc725.js` library and [LSP12](../../standards/universal-profile/lsp12-issued-assets/) JSON schema to encode the data keys related to `LSP12IssuedAssets[]`.
+After setting up the array of assets, you can use the [`erc725.js`](../../tools/erc725js/getting-started.md) library and the [LSP12 JSON Schema](../../standards/universal-profile/lsp12-issued-assets/) to encode the data keys. There are two ways to encode data:
```
Comment (CJ42) [inline docs/learn/dapp-developer/register-lsp12-issued-assets.md:138]:
```suggestion
After setting up the array of assets, you can use the [`erc725.js`](../../tools/erc725js/getting-started.md) library to encode the [`LSP12IssuedAssets[]`](../../standards/universal-profile/lsp12-issued-assets/) data keys. There are two ways to encode data:
```

File: docs/learn/dapp-developer/register-lsp12-issued-assets.md [hunk 3]
```diff
@@ -125,17 +125,35 @@ const issuedAssets = [
...
+
+- **Set the full issued assets list**: If you want to initially set issued assets on a Universal Profile or re-set all elements, you can encode the data without defining optional length or index parameters. Therefore, the issued assets will only consist of the asset addresses you provide in your array of assets.
+- **add, update, or remove existing issued assets**: If the Universal Profile already has issued assets, and you want to add, update, or remove certain assets, you can provide the `startingIndex` and `totalArrayLength` parameters. Therefore, your prepared array of assets only represents a subset of the total issued assets set on the contract.
```
Comment (CJ42) [inline docs/learn/dapp-developer/register-lsp12-issued-assets.md:141]:
```suggestion
- **add, update, or remove existing issued assets**: If the Universal Profile already has issued assets, and you want to add, update, or remove certain assets, you can provide the `startingIndex` and `totalArrayLength` parameters. Therefore, your prepared array only represents a subset of all the issued assets listed in your Universal Profile.
```

File: docs/learn/dapp-developer/register-lsp12-issued-assets.md [hunk 4]
```diff
@@ -191,30 +259,41 @@ const myUPContract = new ethers.Contract(
...
 </Tabs>
 
 ## Set data batch
 
-Next, use the `setDataBatch(...)` function of the Universal Profile to update multiple data keys.
+Next, use the `setDataBatch(...)` function of the Universal Profile to initially set or update multiple data keys.
 
 <Tabs groupId="web3-lib">
-  <TabItem value="web3js" label="web3.js">
+    <TabItem value="ethersjs" label="ethers.js">
```
Comment (CJ42) [inline docs/learn/dapp-developer/register-lsp12-issued-assets.md:282]:
```suggestion
    <TabItem value="ethersjs" label="ethers.js" default>
```

File: docs/learn/expert-guides/universal-profile/register-lsp12-issued-assets.md
Comment (CJ42) [inline docs/learn/expert-guides/universal-profile/register-lsp12-issued-assets.md:1]: why is there the same guide twice?

File: docs/learn/expert-guides/universal-profile/register-lsp12-issued-assets.md [hunk 5]
```diff
@@ -178,21 +195,71 @@ const { keys: lsp12DataKeys, values: lsp12Values } = erc725.encodeData([
...
-## Instantiate the Universal Profile contract
+  </TabItem>
+  <TabItem value="update-element" label="Add, Update, or Remove Issued Assets">
 
-Create an instance of the Universal Profile contract as shown below:
+:::success Version Compatibility
 
-<Tabs groupId="web3-lib">
-  <TabItem value="web3js" label="web3.js">
+Updating a subset of storage elements in an array requires dynamic encoding, which is only available for [`erc725.js`](../../../tools/erc725js/getting-started.md) of version 0.24.0 and above.
```
Comment (CJ42) [inline docs/learn/expert-guides/universal-profile/register-lsp12-issued-assets.md:203]: I don't think we need to mention this. People will always the latest version of the _erc725.js_ library. Suggest removing

**Review by CJ42 — CHANGES_REQUESTED** [review]

**Comment by fhildeb** [comment]
- Applied Suggestions
- Removed similar backend guide

LGTM

**Review by CJ42 — APPROVED** [review]

**Pull request merged**